Source/NSOrderedSet.m

```diff
@@ -0,0 +1,1763 @@
+                            passingTest:(GSPredicateBlock)predicate
+{
+  /* TODO: Concurrency. */
+  NSMutableIndexSet     *set = [NSMutableIndexSet indexSet];
+  BLOCK_SCOPE BOOL      shouldStop = NO;
+  id<NSFastEnumeration> enumerator = self;
+  NSUInteger            count = 0;
+  BLOCK_SCOPE NSLock    *setLock = nil;
+
+  /* If we are enumerating in reverse, use the reverse enumerator for fast
+   * enumeration. */
+  if (opts & NSEnumerationReverse)
+    {
+      enumerator = [self reverseObjectEnumerator];
+    }
+  if (opts & NSEnumerationConcurrent)
+    {
+      setLock = [NSLock new];
+    }
+  {
+    GS_DISPATCH_CREATE_QUEUE_AND_GROUP_FOR_ENUMERATION(enumQueue, opts)
+    FOR_IN (id, obj, enumerator)
+#     if __has_feature(blocks) && (GS_USE_LIBDISPATCH == 1)
+
+      dispatch_group_async(enumQueueGroup, enumQueue, ^(void){
+        if (shouldStop)
+        {
+	  return;
+        }
+        if (predicate(obj, count, &shouldStop))
+        {
+	  [setLock lock];
+	  [set addIndex: count];
+	  [setLock unlock];
+        }
+      });
+#     else
+      if (CALL_BLOCK(predicate, obj, count, &shouldStop))
+        {
+	  /* TODO: It would be more efficient to collect an NSRange and only
+	   * pass it to the index set when CALL_BLOCK returned NO. */
+	  [set addIndex: count];
+        }
+#     endif
+      if (shouldStop)
+        {
+	  break;
+        }
+      count++;
+    END_FOR_IN(enumerator)
+    GS_DISPATCH_TEARDOWN_QUEUE_AND_GROUP_FOR_ENUMERATION(enumQueue, opts);
+  }
+  RELEASE(setLock);
+  return set;
+}
+
+- (NSEnumerator *) objectEnumerator
+{
+  return nil;
+}
+
+- (NSEnumerator *) reverseObjectEnumerator
+{
+  return nil;
+}
+
+- (NSOrderedSet *)reversedOrderedSet
+{
+  return nil;
+}
+
+- (void) getObjects: (__unsafe_unretained id[])aBuffer
+              range: (NSRange)aRange
+{
+}
+
+// Key value coding support
+- (void) setValue: (id)value forKey: (NSString*)key
+{
+}
+
+- (id) valueForKey: (NSString*)key
+{
+    NSEnumerator *e = [self objectEnumerator];
+  id object = nil;
+  NSMutableSet *results = [NSMutableSet setWithCapacity: [self count]];
+
+  while ((object = [e nextObject]) != nil)
+    {
+      id result = [object valueForKey: key];
+
+      if (result == nil)
+        continue;
+
+      [results addObject: result];
+    }
+  return results;
+}
+
+// Key-Value Observing Support
+/*
+- addObserver:forKeyPath:options:context:
+- removeObserver:forKeyPath:
+- removeObserver:forKeyPath:context:
+*/
+- (NSUInteger)_countForObject: (id)object  // required override...
+{
+  return 1;
+}
+
+// Comparing Sets
+- (BOOL) isEqualToOrderedSet: (NSOrderedSet *)aSet
+{
+  if ([self count] != [aSet count])
+    return NO;
+  else
+    {
+      id o, e = [self objectEnumerator];
+
+      while ((o = [e nextObject]))
+        {
+	  if (![aSet containsObject: o])
+            {
+	      return NO;
+            }
+         else
+           {
+             if ([self _countForObject: o] != [aSet _countForObject: o])
```

Comment:
Actually, this is entirely wrong.  In order to be equal the set must also be in the same sequence.

Reply:
I have changed it to check if they are equal and then check if the receiver is a subset of aSet.   This should be sufficient since we are simply checking if it is a strict subset of the other and order matters.